# Netflix EDA Project

This project will explore the Netflix Titles dataset to uncover insights and trends about the types of content available, country-wise production, year-wise additions, and more.

Both **pandas** and **SQL (SQLite)** are used for data analysis, and visualizations using **Matplotlib** and **Seaborn**.

**Source:** (https://www.kaggle.com/datasets/shivamb/netflix-shows)


## Step 1: Clean the Data

Read the dataset:

In [2]:
import pandas as pd

df = pd.read_csv('../data/raw_netflix_titles.csv')


Check the properties:

In [4]:
print(df.shape)
df.info()
df.head()
df.tail()
df.dtypes
df.isnull().sum().sort_values(ascending=False)
df.duplicated().sum()
for col in df.columns:
    print(f"{col}: {df[col].nunique()} unique values")


(8807, 12)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB
show_id: 8807 unique values
type: 2 unique values
title: 8807 unique values
director: 4528 unique values
cast: 7692 unique values
country: 748 unique values
date_added: 1767 unique values
release_year: 74 unique values
rating: 17 unique valu

**Findings:**

Raw netflix titles dataset contains 12 columns and 8807 rows.

Dataset contains 11 object colums and 1 integer column.

Columns director, cast, country, date_added, rating and duration has missing values.

Show_id and title has all unique data so there are no duplicates.

Column durations has a mixed format.

Date_added has string values and should be converted to a datetime object.


Clean the data:

In [9]:
df['director'] = df['director'].fillna('Unknown')
df['cast'] = df['cast'].fillna('Unknown')
df['country'] = df['country'].fillna('Unknown')
df['date_added'] = df['date_added'].fillna('Unknown')
df['rating'] = df['rating'].fillna('Unknown')
df['duration'] = df['duration'].fillna('Unknown')

df['date_added'] = pd.to_datetime(
    df['date_added'].astype(str).str.strip(),
    format='mixed',
    errors='coerce'
)

df['duration_int'] = df['duration'].str.extract(r'(\d+)').astype(float)
df['duration_type'] = df['duration'].str.extract(r'([a-zA-Z]+)')

df['year_added'] = df['date_added'].dt.year
df['month_added'] = df['date_added'].dt.month

cols_to_clean = ['type', 'rating', 'country', 'listed_in']
for col in cols_to_clean:
    df[col] = df[col].str.strip()

Save the clean data:

In [12]:
df.to_csv('../data/cleaned_netflix_titles.csv', index=False)